LibVLCSharp/Shared/Core.cs

```diff
@@ -64,73 +67,114 @@ static void InitializeAndroid()
+                Load(appExecutionDirectory);
+                return;
+            }
 
-                var librariesFolder = Path.Combine(appExecutionDirectory, Constants.LibrariesRepositoryFolderName, arch);
+            var paths = ComputeLibVLCSearchPaths();
 
-                _libvlccoreHandle = PreloadNativeLibrary(librariesFolder, $"{Constants.CoreLibraryName}.dll");
-                
-                if(_libvlccoreHandle == IntPtr.Zero)
-                {
-                    throw new VLCException($"Failed to load required native library {Constants.CoreLibraryName}.dll");
-                }
+            foreach(var path in paths)
+            {
+                if (Load(path, shouldThrow: false)) break;
+            }
 
-                _libvlcHandle = PreloadNativeLibrary(librariesFolder, $"{Constants.LibraryName}.dll");
-                
-                if(_libvlcHandle == IntPtr.Zero)
-                {
-                    throw new VLCException($"Failed to load required native library {Constants.LibraryName}.dll");
-                }
+            if(!Loaded)
+            {
+                throw new VLCException($"Failed to load required native libraries");
             }
-            else if (PlatformHelper.IsMac)
+        }
+
+        static List<string> ComputeLibVLCSearchPaths()
+        {
+            var paths = new List<string>();
+
+#if !NETSTANDARD1_1
+            // compute paths
+            var appExecutionDirectory = Path.GetDirectoryName(typeof(LibVLC).Assembly.Location);
+            var arch = PlatformHelper.IsX64BitProcess ? ArchitectureNames.Win64 : ArchitectureNames.Win86;
+            var librariesFolder = Path.Combine(appExecutionDirectory, Constants.LibrariesRepositoryFolderName, arch);
+            paths.Add(librariesFolder);
+#endif
+            // TODO: add more path
+            return paths;
+        }
+
+        static bool Load(string appExecutionDirectory, bool shouldThrow = true, bool log = true)
+        {
+            if (string.IsNullOrEmpty(appExecutionDirectory)) return false;
+
+            Debug.Assert(PlatformHelper.IsWindows || PlatformHelper.IsMac); // for now
+
+            bool loadResult;
+
+            // no need to load libvlccore on mac currently
+            if (PlatformHelper.IsWindows)
             {
-                _libvlcHandle = PreloadNativeLibrary(appExecutionDirectory, $"{Constants.LibraryName}.dylib");
-                if (_libvlcHandle == IntPtr.Zero)
+                var libvlccorePath = Path.Combine(appExecutionDirectory, $"{Constants.CoreLibraryName}{LibraryExtension}");
+                loadResult = LoadNativeLibrary(libvlccorePath, out _libvlccoreHandle);
+                if (!loadResult) 
                 {
-                    throw new VLCException($"Failed to load required native library {Constants.LibraryName}.dylib");
+                    if (log)
+                        Log($"Failed to load required native library {libvlccorePath}");
```

Comment:
Don't know if it is a good thing to log each failed attempt. It makes the logs harder to read because of the additional (false) failures.
Maybe it is clearer to have a single failure log with all the search locations.